packages/model-viewer/src/features/scene-graph.ts

```diff
@@ -141,6 +151,25 @@ export const SceneGraphMixin = <T extends Constructor<ModelViewerElementBase>>(
         texture.format = RGBFormat;
       }
 
+      if (src instanceof HTMLCanvasElement) {
+        const canvas = src;
+        return new ModelViewerTexture(
+            this[$getOnUpdateMethod](), texture, null, null, {
+              name: 'canvas',
+              // lazily set the canvas dataURL as the "uri"
+              get uri() {
+                // invoke own setter with dataURL
+                this.uri = canvas.toDataURL();
+                return this.uri;
+              },
+              set uri(uri) {
+                // delete getter and setter and set "uri" as a regular
+                // property
+                delete this.uri;
```

Comment:
Huh, that's clever. I don't have any problem with this so long as it works. What's your plan for testing? Particularly, I'd like to ensure that `exportGLB` behaves sanely.

Reply:
I've been testing with the "Export GLB" example in the docs and then loading the exported .glb into the editor to verify that the texture created from canvas was exported correctly and that the model can be validated without issues.
Everything works as expected but the getter/setter aren't invoked at all. In fact, the exported glb is the same even if the gltfImage given to the `ModelViewerTexture` constructor is `null`.
This makes sense to me considering that that the glb export is generated using THREE's `GLTFExporter` which I suspect does not interface with these gltfImages at all.
Since this is the case, what should my litmus test be to make sure that this lazy memoized getter pattern is not breaking anything?

```
  async function exportGLB(){
    let modelViewer = document.getElementById("static-model");

    const canvas = document.createElement('canvas')
    canvas.width = 1000
    canvas.height = 1000
    const ctx = canvas.getContext('2d')
    ctx.fillStyle = 'red'
    ctx.fillRect(0,0,canvas.width,canvas.height)
    const texture = await modelViewer.createTexture(canvas);
    const [material] = modelViewer.model.materials;
    material.pbrMetallicRoughness.baseColorTexture.setTexture(texture);
    
    const glTF = await modelViewer.exportScene();
    var file = new File([glTF], "export.glb");
    var link = document.createElement("a");
    link.download =file.name;
    link.href = URL.createObjectURL(file);
    link.click();
  }
```

Reply:
Ah, fair enough; in that case I'm not particularly worried about it. But yes, I like the test outlined above; maybe just draw an image to the canvas and then verify that the output glb has a reasonable file size. That will give me confidence the data has flowed through.